llvm-build/llvm.ps1

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+. (Join-Path $PSScriptRoot "utils.ps1")
+
+# Build
+
+$srcRoot = Get-RepoRoot
+$buildDir = Join-Path $srcRoot llvm-build
+$llvmCmakeFile = Join-Path $buildDir llvm.cmake
+$llvmRootDir = Join-Path $buildDir llvm-project
+$llvmBuildDir = Join-Path $llvmRootDir build
+$llvmDir = Join-Path $llvmRootDir llvm
+
+if (Test-CommandExists sccache) {
+    Write-Vso "Found sccache command"
+    # Set cap and make sure dir is created
+    $Env:SCCACHE_CACHE_SIZE = "2G"
+    Write-Vso "Starting sccache server"
+    & { sccache --start-server } -ErrorAction SilentlyContinue
+    Write-Vso "Started sccache server"
+}
+elseif (Test-CommandExists ccache) {
+    Write-Vso "Found ccache command"
+
+    if (-not (Test-Path env:\CCACHE_DIR)) {
+        # CCACHE_DIR needs to be set, get the value from ccache
+        $Env:CCACHE_DIR = exec { ccache -k cache_dir }
+    }
+    Assert (![string]::IsNullOrWhiteSpace($Env:CCACHE_DIR)) "CCACHE_DIR is not set"
+
+    # Set cap and make sure dir is created
+    if (!(Test-Path $Env:CCACHE_DIR)) {
+        mkdir $Env:CCACHE_DIR | Out-Null
+    }
+    $Env:CCACHE_DIR = Resolve-Path $Env:CCACHE_DIR
+    ccache -M 2G
+
+    Write-Vso "Found ccache config:"
+    ccache --show-config
+
+    $cacheRoot = $Env:CCACHE_DIR
+
+    Assert (Test-Path $cacheRoot) "$($cacheRoot) is missing"
+    Write-Vso "Using CCACHE_DIR: $($Env:CCACHE_DIR)"
+}
+else {
+    Write-Vso "Did not find ccache command"
+}
+
+
+# Download the current tag of the llvm-project
+$llvmTag = Get-LlvmTag
+exec -wd $buildDir {
+    git clone --depth 1 -b $llvmTag --recurse-submodules --shallow-submodules https://github.com/llvm/llvm-project
+}
+
+Assert (![string]::IsNullOrWhiteSpace($Env:PKG_NAME)) "PKG_NAME is not set"
+
+Assert (Test-CommandExists "cmake") "CMAKE not found"
+Assert (Test-CommandExists "ninja") "Ninja-Build not found"
+if ($IsLinux) {
+    Assert (Test-CommandExists "docker") "Docker not found"
+}
+
+if (!(Test-Path $llvmBuildDir)) {
+    mkdir $llvmBuildDir | Out-Null
+}
+
+Write-Vso "Generating package: $($Env:PKG_NAME)"
+
+if ($IsLinux) {
+    function Build-ContainerImage {
+
+        Write-Vso "Building container image manylinux-llvm-builder"
+        exec -wd $buildDir {
+            $userName = [Environment]::UserName
+            $userId = $(id -u)
+            $groupId = $(id -g)
+            Write-Vso "Get-Content manylinux.Dockerfile | docker build -t manylinux-llvm-builder --build-arg USERNAME=$userName --build-arg USER_UID=$userId --build-arg USER_GID=$groupId --build-arg LLVM_BUILD_DIR=""$llvmBuildDir"" -" "command"
```

Comment:
It is suspicious that the `$userName`, `$userId`, etc. are not surrounded with extra quotes (`"`), but `"$llvmBuildDir"` _is_.